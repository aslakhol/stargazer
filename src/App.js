import React from 'react';
import { connect } from 'react-redux';
import {
  Container,
  Row,
  Col,
} from 'reactstrap';
import ModalExample from './components/ModalExample';
import CardExample from './components/CardExample';
import { simpleAction } from './actions/simpleAction';

class App extends React.Component {
  simpleAction = (event) => {
    this.props.simpleAction();
  }

  constructor(props) {
    super(props);

    this.toggle = () => this.toggle();
    this.state = {
      isOpen: false
    };
  }
  toggle() {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }
  render() {
    return (
      <div>
        <Container>
          <Row>
            <Col md="3" sm="4" mb="3">
              <CardExample className="mb-3" />
            </Col>
            <Col md="3" sm="4" mb="3">
              <CardExample />
            </Col>
          </Row>
          <Row>
            <Col md="3" sm="4" mb="3">
              <CardExample />
            </Col>
            <Col md="3" sm="4" mb="3">
              <CardExample />
            </Col>
          </Row>
        </Container>
        <ModalExample buttonLabel="hei" />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  ...state
})

const mapDispatchToProps = dispatch => ({
  simpleAction: () => dispatch(simpleAction())
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
