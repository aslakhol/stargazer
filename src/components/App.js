import React from 'react';
import {
  Container,
  Col,
  Row,
} from 'reactstrap';

import SearchPageContainer from '../containers/SearchPageContainer';
import SearchBarContainer from '../containers/SearchBarContainer';
import SearchHistoryContainer from '../containers/SearchHistoryContainer';
import FilterButtonsContainer from '../containers/FilterButtonsContainer';
import PersonModalContainer from '../containers/PersonModalContainer';
import PageSelectContainer from '../containers/PageSelectContainer';
import SortingButtonsContainer from '../containers/SortingButtonsContainer';

const App = () => (
  <Container>
    <Row className="mt-3">
      <Col>
        <FilterButtonsContainer />
      </Col>
      <Col>
        <SortingButtonsContainer />
      </Col>
    </Row>
    <Row className="my-3 mx-0">
      <SearchBarContainer />
    </Row>
    <Row className="mb-3">
      <SearchPageContainer />
    </Row>
    <Row>
      <PageSelectContainer />
    </Row>
    <Row>
      <SearchHistoryContainer />
    </Row>
    <PersonModalContainer />
  </Container>
);

export default App;
