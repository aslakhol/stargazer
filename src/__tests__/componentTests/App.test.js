import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../../components/App';

describe('<App />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const component = shallow(<App persons={[]} query="" fetchPersonsIfNeeded={() => true} />);

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
