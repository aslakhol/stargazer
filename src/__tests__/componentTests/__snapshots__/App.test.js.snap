// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() renders the component 1`] = `
<Container
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      className="titleStyle"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      STARGAZER
    </Col>
  </Row>
  <Row
    className="mt-3"
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Connect(FilterButtons) />
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Connect(SortButtons) />
    </Col>
  </Row>
  <Row
    className="my-3 mx-0"
    tag="div"
  >
    <Connect(SearchBar) />
  </Row>
  <Row
    className="mb-3"
    tag="div"
  >
    <Connect(SearchPage) />
  </Row>
  <Row
    tag="div"
  >
    <Connect(PageSelect) />
  </Row>
  <Row
    tag="div"
  >
    <Connect(SearchHistory) />
  </Row>
  <Connect(PersonModal) />
</Container>
`;
